api/v1alpha1/serviceintentions_types.go

```diff
@@ -0,0 +1,220 @@
+}
+
+func (in *ServiceIntentions) SetSyncedCondition(status corev1.ConditionStatus, reason, message string) {
+	in.Status.Conditions = Conditions{
+		{
+			Type:               ConditionSynced,
+			Status:             status,
+			LastTransitionTime: metav1.Now(),
+			Reason:             reason,
+			Message:            message,
+		},
+	}
+}
+
+func (in *ServiceIntentions) SyncedCondition() (status corev1.ConditionStatus, reason, message string) {
+	cond := in.Status.GetCondition(ConditionSynced)
+	if cond == nil {
+		return corev1.ConditionUnknown, "", ""
+	}
+	return cond.Status, cond.Reason, cond.Message
+}
+
+func (in *ServiceIntentions) SyncedConditionStatus() corev1.ConditionStatus {
+	condition := in.Status.GetCondition(ConditionSynced)
+	if condition == nil {
+		return corev1.ConditionUnknown
+	}
+	return condition.Status
+}
+
+func (in *ServiceIntentions) ToConsul(datacenter string) api.ConfigEntry {
+	return &capi.ServiceIntentionsConfigEntry{
+		Kind:    in.ConsulKind(),
+		Name:    in.Spec.Destination.Name,
+		Sources: in.Spec.Sources.toConsul(),
+		Meta:    meta(datacenter),
+	}
+}
+
+func (in SourceIntentions) toConsul() []*capi.SourceIntention {
+	var consulSourceIntentions []*capi.SourceIntention
+	for _, intention := range in {
+		consulSourceIntentions = append(consulSourceIntentions, intention.toConsul())
+	}
+	return consulSourceIntentions
+}
+
+func (in *ServiceIntentions) ConsulGlobalResource() bool {
+	return false
+}
+
+func (in *SourceIntention) toConsul() *capi.SourceIntention {
+	if in == nil {
+		return nil
+	}
+	return &capi.SourceIntention{
+		Name:        in.Name,
+		Namespace:   in.Namespace,
+		Action:      in.Action.toConsul(),
+		Description: in.Description,
+	}
+}
+
+func (in IntentionAction) toConsul() capi.IntentionAction {
+	return capi.IntentionAction(in)
+}
+func (in *ServiceIntentions) MatchesConsul(candidate api.ConfigEntry) bool {
+	configEntry, ok := candidate.(*capi.ServiceIntentionsConfigEntry)
+	if !ok {
+		return false
+	}
+	// No datacenter is passed to ToConsul as we ignore the Meta field when checking for equality.
+	return cmp.Equal(
+		in.ToConsul(""),
+		configEntry,
+		cmpopts.IgnoreFields(capi.ServiceIntentionsConfigEntry{}, "Namespace", "Meta", "ModifyIndex", "CreateIndex"),
+		cmpopts.IgnoreFields(capi.SourceIntention{}, "LegacyID", "LegacyMeta", "LegacyCreateTime", "LegacyUpdateTime", "Precedence", "Type"),
+		cmpopts.IgnoreUnexported(),
+		cmpopts.EquateEmpty(),
+	)
+}
+
+func (in *ServiceIntentions) Validate() error {
+	var errs field.ErrorList
+	path := field.NewPath("spec")
+	for i, source := range in.Spec.Sources {
+		if err := source.Action.validate(path.Child("sources").Index(i)); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if len(errs) > 0 {
+		return apierrors.NewInvalid(
+			schema.GroupKind{Group: ConsulHashicorpGroup, Kind: common.ServiceIntentions},
+			in.KubernetesName(), errs)
+	}
+	return nil
+}
+
+func (in *ServiceIntentions) Default() {
```

Comment:
```suggestion
// Default sets zero value fields on this object to their defaults.
func (in *ServiceIntentions) Default() {
```

Should this be `SetDefaults()`? Or is `Default()` the idiomatic name?

Reply:
Default() is idiomatic and is in the similar vein as Validate()